src/Stratis.Bitcoin.Features.Consensus/StakeValidator.cs

```diff
@@ -204,7 +208,7 @@ public void CheckProofOfStake(ContextStakeInformation context, ChainedHeader pre
             }
 
             // Min age requirement.
-            if (this.IsConfirmedInNPrevBlocks(prevUtxo, prevChainedHeader, this.consensusOptions.StakeMinConfirmations - 1))
+            if (this.IsConfirmedInNPrevBlocks(prevUtxo, prevChainedHeader, this.consensusOptions.GetStakeMinConfirmations(prevChainedHeader.Height + 1, this.network) - 1))
```

Comment:
How about we put the `StakeMinConfirmations` as a parameter in the caller of this method, then it will be calculated in the rule itself, the rules can be network specific. would that work?

Reply:
We will need this only when StakeValidator goes into a rule. 
And when it goes it will be refactored anyway so I don't see a reason why would we want to do it in this PR?

Reply:
`StakeValidator` will likely not go in to a rule on its own its rather a library called by existing rules (but not sure yet at what extent)
I am fine with keeping this as is for now, not to bulk the PR.

Reply:
Ok 👍